data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchAccessor.java

```diff
@@ -55,20 +58,20 @@
 import static org.opensearch.dataprepper.plugins.source.opensearch.worker.client.model.MetadataKeyAttributes.DOCUMENT_ID_METADATA_ATTRIBUTE_NAME;
 import static org.opensearch.dataprepper.plugins.source.opensearch.worker.client.model.MetadataKeyAttributes.INDEX_METADATA_ATTRIBUTE_NAME;
 
-public class OpenSearchAccessor implements SearchAccessor, ClusterClientFactory {
+public class OpenSearchAccessor implements SearchAccessor, ClusterClientFactory<OpenSearchClient> {
 
     private static final Logger LOG = LoggerFactory.getLogger(OpenSearchAccessor.class);
 
     static final String PIT_RESOURCE_LIMIT_ERROR_TYPE = "rejected_execution_exception";
     static final String INDEX_NOT_FOUND_EXCEPTION = "index_not_found_exception";
     static final String SCROLL_RESOURCE_LIMIT_EXCEPTION_MESSAGE = "Trying to create too many scroll contexts";
 
-    private final OpenSearchClient openSearchClient;
+    private final OpenSearchClientRefresher openSearchClientRefresher;
```

Comment:
Similar to my comment above, make this: `Supplier<OpenSearchClient>` to decouple getting the client from the fact that it is refreshed.